File: 1-linear_regression/linearregression.jl
using Plots

default(framestyle=:box,grid=false,legend=false)

m = 0.3
b = 5
linear_function(x) = m*x.+b

x=range(0,10,length=100)
y=linear_function.(x)
y_noisy=@. linear_function.(x)+0.3*randn()

plot(x,y,xlims=(minimum(x)-1,maximum(x)+1),linewidth=2)
plot!(x,y_noisy,seriestype=:scatter,markerstrokecolor=:auto)
savefig("1-linear_regression\\jllinearregression.png")
display(Plots.plot!())
wait()
exit()

File: 1-linear_regression/linearregression.py
import matplotlib.pyplot as pyplot
import numpy as np
import os
from datetime import datetime

noise = 3*np.random.randn(50)
m = 2
b = 3

x = np.linspace(0,10,50)
y = m*x+b+noise

pyplot.scatter(x,y)
pyplot.plot(x,m*x+b,linewidth=2,color='red')
pyplot.margins(x=0.1,y=0.1)
if os.name=="nt":
    pyplot.savefig(f"plot{datetime.timestamp(datetime.now())}.png")
else:
    pyplot.savefig(f"plot{datetime.timestamp(datetime.now())}.png")
pyplot.show()
